src/Security/SecProtocolMetadata.cs

```diff
@@ -51,9 +51,181 @@ public class SecProtocolMetadata : NativeObject {
 
 		public bool EarlyDataAccepted => sec_protocol_metadata_get_early_data_accepted (GetCheckedHandle ()) != 0;
 
-		//
-		// MISSING: all the block APIs
-		//
+ 		[DllImport (Constants.SecurityLibrary)]
+		[return: MarshalAs (UnmanagedType.I1)]
+		extern static bool sec_protocol_metadata_challenge_parameters_are_equal (IntPtr metadataA, IntPtr metadataB);
 
+		public static bool ChallengeParametersAreEqual (SecProtocolMetadata metadataA, SecProtocolMetadata metadataB)
+		{
+			if (metadataA == null)
+				throw new ArgumentNullException (nameof (metadataA));
+			if (metadataB == null)
+				throw new ArgumentNullException (nameof (metadataB));
+			return sec_protocol_metadata_challenge_parameters_are_equal (metadataA.GetCheckedHandle (), metadataB.GetCheckedHandle ());
+		}
+
+ 		[DllImport (Constants.SecurityLibrary)]
+		[return: MarshalAs (UnmanagedType.I1)]
+		extern static bool sec_protocol_metadata_peers_are_equal (IntPtr metadataA, IntPtr metadataB);
+
+		public static bool PeersAreEqual (SecProtocolMetadata metadataA, SecProtocolMetadata metadataB)
+		{
+			if (metadataA == null)
```

Comment:
Same: what if both inputs are null?

Reply:
I checked the API and when both are NULL the API returns false. I'll copy the behaviour.